### PR TITLE
Stop logs safely in the block logger and asynchronously where possible

### PR DESCRIPTION
**Pull request by andyp1per**

### CODE REVIEW

**Comment by amilcarlucas** [comment]
I recently had a power issue on one of my drones. because the flushing is happening often I was able to see the vcc going down.

If I understand correctly this will make flushing less often, causing sometimes the last few precious miliseconds to not get logged?

**Comment by andyp1per** [comment]
@amilcarlucas - no this should only affect logs when you disarm.

**Review by tridge — APPROVED** [review]

**Comment by tridge** [comment]
@peterbarker merge when you are happy with this

File: libraries/AP_Logger/AP_Logger_Backend.cpp [hunk 1]
```diff
@@ -57,7 +57,7 @@ void AP_Logger_Backend::periodic_1Hz()
     if (_rotate_pending && !logging_enabled()) {
         _rotate_pending = false;
         // handle log rotation once we stop logging
-        stop_logging();
+        stop_logging_eventually();
```
Comment (tridge) [inline libraries/AP_Logger/AP_Logger_Backend.cpp:60]: maybe stop_logging_async() ?

Comment (andyp1per) [reply libraries/AP_Logger/AP_Logger_Backend.cpp]: Done

**Comment by peterbarker** [comment]
Merged, thanks!

**Pull request merged**